File: El Mamoun/Lv 1/Problem 4.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
/*
 Write a program to ask the user to enter his/her: • Age • Driver license
 Then Print “Hired” if his\her age is grater than 21 and s/he has a driver license, otherwise Print “Rejected”
*/

namespace Problem_4
{
    internal class Program
    {
        struct stInfo
        {
            public short Age;
            public bool HasDriveLicence;
        }
        static stInfo ReadPersonInfo()
        {
            stInfo Info;
            Console.Write("Please Enter Your Age?: ");
            Info.Age = Convert.ToInt16(Console.ReadLine());

            Console.Write("Do you have DriveLicence?: ");
            Info.HasDriveLicence = Convert.ToBoolean(Console.ReadLine());

            return Info;
        }
        static bool IsAccepted(stInfo Info)
        {
            return (Info.Age > 21 && Info.HasDriveLicence);

        }
        static void PrintRestult(stInfo Info)
        {
            if (IsAccepted(Info))
                Console.WriteLine("Hired");
            else
                Console.WriteLine("Rejected");
        }


        static void Main(string[] args)
        {

            PrintRestult(ReadPersonInfo());

            Console.ReadKey();
        }
    }
}

File: El Mamoun/Lv 1/Problem 4.cpp
#include<iostream>
using  namespace std;

/*
 Problem 4 :Write a program to ask the user to enter his/her: • Age • Driver license
 Then Print “Hired” if his\her age is grater than 21 and s/he has a driver license, otherwise Print “Rejected”
*/

struct stPersonInfo 
{
	short age = 0;
	bool HasDeriverLicence;
};

stPersonInfo ReadPersonInfo()
{
	stPersonInfo Info;

	cout << "Please Enter Your Age? ";
	cin >> Info.age;

	cout << "Do you have Drive Licence? ";
	cin >> Info.HasDeriverLicence;

	return Info;
}

bool IsAccecpted(stPersonInfo  Info)
{

	return(Info.age > 21 && Info.HasDeriverLicence);
		
}

void PrintResult(stPersonInfo Info)
{

	if (IsAccecpted(Info))
		cout << "Your Hired \n";
	else
		cout << "Your Rejected \n";

}


int main()
{
	PrintResult(ReadPersonInfo());

	return 0;
}